tests/rbd/test_rbd_immutable_cache.py

```diff
@@ -0,0 +1,225 @@
+"""Test case covered -
+    CEPH-83575581 - Configuring the ceph-immutable-object-cache daemon
+
+    Pre-requisites :
+    1. Cluster must be up and running with capacity to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create RBD based pool and an Image
+    2. Enable the immutable cache client settings
+    3. Install the ceph-immutable-object-cache package
+    4. Create a unique Ceph user ID, the keyring
+    5. Enable the ceph-immutable-object-cache daemon with created client
+"""
+import time
+
+from ceph.utils import get_node_by_id
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+ceph_conf_path = "/etc/ceph/ceph.conf"
+immutable_cache_path = "/mnt/immutable-cache/"
+
+
+def execute_command(node, command):
+    result = node.exec_command(cmd=command, sudo=True)
+    if result[1]:
+        log.error(f"Error executing command: {command}\n{result[1]}")
+    return result[0]
+
+
+def configure_immutable_cache(rbd_obj, client_node, user_name):
+    """Method to configure immutable object cache.
+
+    Args:
+        rbd_obj: RBD object
+        client_node: cache client node
+        user_name: name of the client user
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+
+    """
+
+    # Update the Ceph client configuration
+    client_config = {
+        "immutable_object_cache_max_size": "10G",
+        "immutable_object_cache_path": "/mnt/immutable-cache/",
```

Comment:
You can just initialise this with "immutable_object_cache_path": immutable_cache_path so that if we need to change this value in future we can just change it in one place.

Reply:
done